src/clj/puppetlabs/services/jruby/jruby_schemas.clj

```diff
@@ -99,19 +106,22 @@
                      (nil? (schema/check JRubyInstanceState @%)))
                'JRubyInstanceState))
 
+(def JRubyInstanceSchema
+  {:pool pool-queue-type
+   :id schema/Int
+   :max-requests schema/Int
+   :flush-instance-fn IFn
+   :state JRubyInstanceStateContainer
+   :scripting-container ScriptingContainer
+   schema/Keyword schema/Any})
```

Comment:
When I run the following code in a REPL with the current definition of `JRubyInstance`....

```clj
(schema/check JRubyInstance
              (map->JRubyInstance
               {:pool (JRubyPool. 10)
                :id 1
                :max-requests 100
                :flush-instance-fn identity
                :state (atom {:borrow-count 1})
                :scripting-container (ScriptingContainer.
                                      LocalContextScope/SINGLETON)
                :foo "bar"}))
```
I get:

```
=> {:foo disallowed-key}
```
However, if I add a map allowing extra key/value pairs to the JRubyInstance definition, like this:

```
(schema/defrecord JRubyInstance
  [pool :- pool-queue-type
   id :- schema/Int
   max-requests :- schema/Int
   flush-instance-fn :- IFn
   state :- JRubyInstanceStateContainer
   scripting-container :- ScriptingContainer]
  {schema/Keyword schema/Any})
```
... and then run the same `schema/check` from above, I now get:

```
=> nil
```
With this, I also still get an error if a required key is missing.  For example:

```clj
(schema/check JRubyInstance
              (map->JRubyInstance
               {:pool (JRubyPool. 10)
                :id 1
                :flush-instance-fn identity
                :state (atom {:borrow-count 1})
                :scripting-container (ScriptingContainer.
                                      LocalContextScope/SINGLETON)
                :foo "bar"}))
```

... would produce ...

```
{:max-requests (not (integer? nil))}
```

This utilizes the `extra-key-schema?` arg to the `defrecord` macro - https://github.com/plumatic/schema/blob/schema-1.0.4/src/cljx/schema/core.cljx#L1112-L1114.

Maybe this would work as an alternative that would allow us to not have to have the separate schema?